Sources/VLCPlaybackController.m

```diff
@@ -872,36 +873,65 @@ - (void)jumpBackward:(int)interval
     [_mediaPlayer jumpBackward:interval];
 }
 
-- (NSString *)screenAspectRatio
+- (UIScreen *)currentScreen
 {
-    UIScreen *screen = [[UIDevice currentDevice] VLCHasExternalDisplay] ? [UIScreen screens][1] : [UIScreen mainScreen];
-    return [NSString stringWithFormat:@"%d:%d", (int)screen.bounds.size.width, (int)screen.bounds.size.height];
+    return [[UIDevice currentDevice] VLCHasExternalDisplay] ? [UIScreen screens][1] : [UIScreen mainScreen];
+}
+
+- (void)switchToFillToScreen
+{
+    UIScreen *screen = [self currentScreen];
+    CGSize screenSize = screen.bounds.size;
+
+    CGSize videoSize = _mediaPlayer.videoSize;
+
+    CGFloat ar = videoSize.width / (float)videoSize.height;
+    CGFloat dar = screenSize.width / (float)screenSize.height;
+
+    CGFloat scale;
+
+    if (dar >= ar) {
+        scale = screenSize.width / (float)videoSize.width;
+    } else {
+        scale = screenSize.height / (float)videoSize.height;
+    }
+
+    // Multiplied by screen.scale in consideration of pt to px
+    _mediaPlayer.scaleFactor = scale * screen.scale;
+    _isInFillToScreen = YES;
```

Comment:
I don't think you need to introduce this state. Can't you just check the scaleFactor on the mediaPlayer?

Reply:
Well, I guess we could but it felt a bit wrong since the `scaleFactor` is not particularly reserved for the FillToScreen feature, it could be used for other things.
At the end of the day, I have a slight preference for the `BOOL` but I don't mind changing it.